chalice/deploy/planner.py

```diff
@@ -93,6 +95,15 @@ def _plan_deletion(self,
                     params={'function_name': resource_values['lambda_arn']},
                 )
                 plan.append(apicall)
+            elif resource_values['resource_type'] == 'iam_role':
+                # TODO: add the role_name to the deployed.json?
```

Comment:
Can you elaborate on this todo?

Reply:
+1. I think it is because we are parsing the name from the role arn, which is not a great practice as there may be edge cases. If it was included in the `deployed.json`, we would not have to worry about including this logic.

Reply:
Yep, I've updated the todo.